### ISSUE TITLE
Rumble is marginally lowered since version 3.1.6

### ISSUE THREAD

**Issue by bmijnendonckx** [Ryochan7/DS4Windows#2603]
Since the latest update (3.1.6) my controller's rumble has been lowered by quite a lot.
My profile settings were at 30% rumble and the game I play, Rocket League, has a vibration setting that is at 1.0 on default.
I have now set my profile's rumble to 100% and my Rocket League setting to the maximum 3.0 and even after increasing every rumble to it's max it doesn't even feel like half of the rumble it used to be.

I have also just updated my controller firmware to see if this fixes anything, and it didn't change anything.

 - Controller Make and Model: Sony DS5 (CFI-ZCT1W)
 - OS: Windows 11 Pro 22H2 build 22621.607
 - DS4Windows Version 3.1.6

I'm not sure if this has been addressed, as I've seen a new rumble system being mentioned and if the only solution is, for me to roll back my DS4 to the last version that worked for me? Which would be 3.1.4.

Edit: Can confirm that running 3.1.4 still works fine

Sorry if this is a duplicate. Thanks for your work!

**Comment by Kanuan** [comment]
This won't bring the old rumble emulation back, but will bring the max strength closer to what it was:

1. Have your DualSense connected to DS4Windows
1. Go to the "Settings" tab
1. Locate the "device options" button and click on it
1. Adjust the DualSense rumble strength level back to "high"
    
-----------------

# An explanion on the "Improved rumble emulation mode"

I was the one that introduced the change from "standard rumble emulation" to "improved rumble emulation". Let me explain the change.

To start, let me differentiate between 2 rumble strengths:

- **Game Rumble Strength (GameRumbStr):** what the game instructs the controller rumble strength to be
- **DualSense Rumble Strength (DualRumbStr):** the "real" strength the controller is actually rumbling on

DualSense has a setting on itself that allows the user to make the DualRumbStr weaker or stronger, the "rumble strength level" which is the setting I've instruced you on changing back to "high".

The problem with that setting is that on the "old" rumble emulation it acts more like a Strength REGION selector instead of it just limiting the "max" strength.

What I mean is, the GameRumbStr from 1-100% would behave on the controller as if it were:

- High: 87,5-100%
- Medium: 48-60%
- Low: 1-12%

The DualSense rumble emulation was MEANT to "mimick" the xbox 360 / DS4 rumble, but because of how it worked before there was no "depth" to it. Games trying to make the controller rumble at 1% or 100% would feel basically the same since the actual rumble strength would vary AT MAX  13%.

Example: Gof of War "rowing the boat" rumble starts softly (Like, 5% rumble strengtrh) and the gets pretty high as Kratos finishes the rowing motion. On the old rumble emulation, all that you felt was a "constant" rumble since the games' "5%" would be 89% on the controller.

The Improved Rumble Emulation, that is a native controller feature introduced in one of the controller's firmware update, makes the controller rumble actually go from 1-100% on high and 1-60% on Medium. It has more depth and is closer to how you'd feel if you were using a DualShock 4.

**test this yourself:** 

1. On DS4Windows v3.1.4, test how much difference you can feel between 1% and 100% when testing rumble on the Profile Editor -> "Other tab"
    - Make sure to bring the DualSense Rumble Strength level back to HIGH before the test by adjusting its settings in the "hidden" Device Options menu
1. Repeat the test on v3.1.6 or higher
1. Report back your impression

## Users that were already using DualSense with DS4Windows before the update need to manually switch to "high" strength level

Because on the Improved Rumble Emulation "medium" is actually quite weaker, I've adjusted the "default" DualSense level strength to "High", which feels closer to a DualShock 4 then before, in which "High" was indeed exaustingly strong.

Unfortunately everyone that was already using a DualSense will still be on medium, requiring manual adjustment of the strength level back to High.

# Planned changes

I intend on:

1. Moving the strength level to the profile editor so it's more visible
1. Have an option to switching back to the "Old" rumble emulation

**Comment by bmijnendonckx** [comment]
Thanks for the detailed explanation and those changes are definitely a good idea as I had no idea that button was there, I totally looked over it.

So I'm currently comparing the rumble between the 2 versions, and I'll add my personal findings/opinion below.

On the old version with the setting on medium, the rumble feels like it's a consistent rumble over the entire controller (or at least the handles which you hold) while the new version set to high feels kinda different. It's hard to explain, but it feels like it not entirely rumbling. I can feel the rumble from the inside, but it doesn't really transfer to the outside casing as well.

I also tried the low settings on the old version and that also feels as I described the medium to be just barely noticeable, but it does transfer to the outside.

Of course, this is all based on what I feel and therefore isn't very technical/scientific. I am going to assume that I am also biased towards the older version's implementation due to it being the way it was for using it on an almost daily basis for quite an extensive amount of time. However, the new implementation does feel more natural and less static.

I feel like with the new version on high compared to the old version on medium, the distance in time between the rumble has changed. The old one being so close to each other that it feels constant while the new one is much further apart giving me a pulsing vibration.

I might give the new vibration a try on some other games, but for Rocket League I will probably stick to the way I am used to.
I look forward to these planned changes, and thanks for the quick reply!

**Comment by Kanuan** [comment]
Another way to test is in a more "real time" way is:

1. Set DualSense device specific Rumble Strength level to HIGH
2. Set the profile rumble strength to 100%
3. Set the profile to emulate a Xbox 360 controller
4. Go to DS4Windows -> Settings tab
5. On the Settings tab, open the Xinput Checker Tool
6. Test the rumble strength on the sliders at the bottom

**Comment by Kanuan** [comment]
I think this should cover all useful cases. This is only a draft for the interface, still need to code the logic for this when I have the time. Not today though

![image](https://user-images.githubusercontent.com/24910442/192113960-e5e54d1a-0650-46d4-9215-450cef4081c5.png)

**Comment by MGxFF** [comment]
what do I do if my controller is not showing up....I already tried disconnecting and also updating.

**Comment by Gobblerpl** [comment]
Hey, I also noticed that the vibration is noticeably weaker. 

Kanuan I checked the vibration power as you said. It is indeed a cool feature previously the vibrations were either there or not there now they have a gradual intensity. 

Unfortunately overall they are too weak. I don't have a comparison to consoles because I've been a PC gamer since forever and dualsense is my first gamepad but before I could feel the vibrations clearly and now they are so soft that I forget about them. 

Thank you for your work on ds4windows you are great :)
